src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long productId) {
+        return Storage.products.stream()
+                .filter(product -> product.getId().equals(productId))
+                .findFirst();
+    }
+
+    @Override
+    public Product update(Product product) {
+        getAllProducts().stream()
+                .filter(prod -> prod.getId().equals(product.getId()))
+                .forEach(prod -> prod = product);
+
+        return product;
```

Comment:
let's pass the new Product Object with the same id and see if the list(storage) will change.

Reply:
done